docs/CONTRIBUTING.md

```diff
@@ -7,6 +7,24 @@ Contributing to NeoForge
 
 3) Follow the code style of the class you're working in (braces on newlines & spaces instead of tabs in Forge classes, inline brackets in patches, etc).
 
+## Workflow
+
+0. Have preliminary discussions on Discord (`#neoforge-github`)
+1. Fork the repository
+2. Check out your fork
+3. Make a branch
+4. Run `gradlew setup` from the project root
```

Comment:
`neoforge:setup` should be enough and avoids needing to decompile the base project too.

Reply:
Will `unpackSourcePatches` incur a decomp of the base project for diff purposes if it doesn't get decompiled during setup?  I'm not sure how it operates under the hood.

Reply:
Think I usually prefix that one with `neoforge:` too, come to think of it.

Reply:
@marchermans would know.  My understanding is that `unpackSourcePatches` generates the patches by doing a source diff between the `base` project and the `neoforge` project, which means that base must be setup as a prereq.

Reply:
No it is not needed.
The patches get generated from the unpatched jar in the neoforge subproject.

The base project is purely informative.

Reply:
Yeah, any modifications on base don't matter, they don't flow into any patches and are overridden the next time.